lib/elixir/src/elixir_lexical.erl

```diff
@@ -40,8 +40,8 @@ trace({import, Meta, Module, Opts}, #{lexical_tracker := Pid}) ->
 
   ?tracker:add_import(Pid, Module, Only, Meta, Imported and should_warn(Meta, Opts)),
   ok;
-trace({alias, Meta, _Old, New, Opts}, #{lexical_tracker := Pid}) ->
-  ?tracker:add_alias(Pid, New, Meta, should_warn(Meta, Opts)),
+trace({alias, Meta, _Old, New, Opts}, #{lexical_tracker := Pid, function := Function}) ->
+  ?tracker:add_alias(Pid, New, Meta, should_warn(Meta, Opts), Function),
```

Comment:
Super massive nitpick: 

```suggestion
  ?tracker:add_alias(Pid, New, Meta, should_warn(Meta, Opts), Function /= nil),
```

So we send less data to the GenServer. :)

Reply:
Too much in my opinion. I feel like I need to undo the `warn` logic when I have to understand its impact in the lexical tracker. While checking for `warn` and `function?` had a clearer meaning to me. WDYT?

Reply:
I'm fine either way. 

I liked the idea of having the emitting part decide what to warn, with atom names to explain, and the receiving part just following instructions.

The downside with the non-bool atom is that I needed to come up with names (not fully satisfied with `:except_unused_shadowed`), and also that we'd probably need guards to make sure we don't emit something else and end up with some random accidental behavior. Finally, consistency with other places.

Will revert to the double bool!

Reply:
[4b835b6](https://github.com/elixir-lang/elixir/pull/13564/commits/4b835b6c40bd33b5d05b4628168ca1dad78c095c)